components/crud-web-apps/jupyter/backend/apps/common/form.py

```diff
@@ -171,6 +187,27 @@ def set_notebook_memory(notebook, body, defaults):
     container = notebook["spec"]["template"]["spec"]["containers"][0]
 
     memory = get_form_value(body, defaults, "memory")
+    memoryLimit = get_form_value(body, defaults, "memoryLimit")
+
+    limit_factor = utils.load_spawner_ui_config()["memory"].get("limitFactor")
+    if not memoryLimit and limit_factor != "none":
+        memoryLimit = str(
+            round((
+                float(memory.replace('Gi', '')) * float(
+                    limit_factor)), 1)) + "Gi"
+        if not memory:
+            memoryLimit = str(round((float(
+                utils.load_spawner_ui_config()["memory"].get(
+                    "value").replace('Gi', '')) * float(
+                        limit_factor)), 1)) + "Gi"
+
+    if memoryLimit is not (None and ''):
```

Comment:
same as above, we could use the inverse check to slightly simplify the code and the indentation below